### PR TITLE
wolfSSH Client

### PR DESCRIPTION
**Pull request by ejohnstown**
Phase 1 of the wolfSSH client.

### CODE REVIEW

File: apps/wolfssh/wolfssh.c [hunk 1]
```diff
@@ -0,0 +1,1089 @@
...
+    appName = basename(appPath);
+    /* Attempt to use the actual program name from the caller. Otherwise,
+     * default to "wolfssh". */
+    if (appName == NULL) {
+        appName = "wolfssh";
+    }
+
+    printf("%s v%s\n", appName, LIBWOLFSSH_VERSION_STRING);
+    printf("usage: %s [-E logfile] [-G] [-l login_name] [-N] [-p port] [-V]\n",
+            appName);
```
[the remark this answers is not in the document]
Comment (JacobBarthelmeh) [reply apps/wolfssh/wolfssh.c:98]: Help menu should show what -V is (verbose debug? version?)

**Review by JacobBarthelmeh — CHANGES_REQUESTED** [review]

File: apps/wolfssh/wolfssh.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (ejohnstown) [reply apps/wolfssh/wolfssh.c:98]: I'm following the options as displayed by ssh. We don't support `-abcdef` strings of options, though. I should have "destination" in there.

File: apps/wolfssh/wolfssh.c [hunk 2]
```diff
@@ -0,0 +1,1089 @@
...
+    }
+    if (config->command) {
+        free(config->command);
+    }
+
+    return 0;
+}
+
+
+THREAD_RETURN WOLFSSH_THREAD client_test(void* args)
```
[the remark this answers is not in the document]
Comment (ejohnstown) [reply apps/wolfssh/wolfssh.c:817]: Done.

File: apps/wolfssh/wolfssh.c [hunk 3]
```diff
@@ -0,0 +1,1089 @@
...
+        int reply, void *ctx)
+{
+    char reqStr[] = "SampleRequest";
+
+    if ((WOLFSSH *)ssh != *(WOLFSSH **)ctx) {
+        printf("ssh(%p) != ctx(%p)\n", ssh, *(WOLFSSH **)ctx);
+        return WS_FATAL_ERROR;
+    }
+
+    if (strlen(reqStr) == sz
```
[the remark this answers is not in the document]
Comment (ejohnstown) [reply apps/wolfssh/wolfssh.c:509]: We need some refactoring of the API compatibility layer soon. The file handling needs an overhaul.

File: apps/wolfssh/wolfssh.c [hunk 4]
```diff
@@ -0,0 +1,1089 @@
...
+        err_sys("client requires a username parameter.");
+
+#ifdef SINGLE_THREADED
+    if (keepOpen)
+        err_sys("Threading needed for terminal session\n");
+#endif
+
+    if (config.keyFile) {
+        ret = ClientSetPrivateKey(config.keyFile);
+        if (ret != 0) {
```
[the remark this answers is not in the document]
Comment (ejohnstown) [reply apps/wolfssh/wolfssh.c]: This should work now.

File: apps/wolfssh/wolfssh.c (hunk 3 above)
[the remark this answers is not in the document]
Comment (ejohnstown) [reply apps/wolfssh/wolfssh.c]: Done.

File: apps/wolfssh/wolfssh.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (ejohnstown) [reply apps/wolfssh/wolfssh.c:99]: Added it to the readme file.

**Comment by JacobBarthelmeh** [comment]
Please resolve this build `./configure --enable-sshclient --disable-term` now that a recent fix for --disable-term was added. (or throw error in configure if that's an invalid build)

**Review by JacobBarthelmeh — APPROVED** [review]

**Pull request merged**